dds/idl/typeobject_generator.cpp

```diff
@@ -526,870 +526,22 @@ operator<<(std::ostream& out, const OpenDDS::XTypes::TypeIdentifier& ti)
   return out;
 }
 
-namespace {
-template<typename T>
-std::string getTypeName();
-
-template<>
-std::string getTypeName<OpenDDS::DCPS::String>()
-{
-  return "String";
-}
-
-template<>
-std::string getTypeName<OpenDDS::XTypes::AppliedAnnotationParameterSeq>()
-{
-  return "XTypes::AppliedAnnotationParameterSeq";
-}
-
-template<>
-std::string getTypeName<OpenDDS::XTypes::AnnotationParameterValue>()
-{
-  return "XTypes::AnnotationParameterValue";
-}
-
-template<>
-std::string getTypeName<OpenDDS::XTypes::AppliedBuiltinMemberAnnotations>()
-{
-  return "XTypes::AppliedBuiltinMemberAnnotations";
-}
-
-template<>
-std::string getTypeName<OpenDDS::XTypes::AppliedAnnotationSeq>()
-{
-  return "XTypes::AppliedAnnotationSeq";
-}
-
-template<>
-std::string getTypeName<OpenDDS::XTypes::AppliedVerbatimAnnotation>()
-{
-  return "XTypes::AppliedVerbatimAnnotation";
-}
-
-template<>
-std::string getTypeName<OpenDDS::XTypes::AppliedBuiltinTypeAnnotations>()
-{
-  return "XTypes::AppliedBuiltinTypeAnnotations";
-}
-
-template<>
-std::string getTypeName<OpenDDS::XTypes::CompleteTypeDetail>()
+void dump_bytes(const OpenDDS::XTypes::TypeObject& to)
 {
-  return "XTypes::CompleteTypeDetail";
-}
-}
-
-template<typename T>
-std::ostream&
-operator<<(std::ostream& out, const OpenDDS::XTypes::Optional<T>& optional)
-{
-  if (optional.present) {
-    out << "XTypes::Optional<" << getTypeName<T>()  << ">(";
-    if (getTypeName<T>() == "String") {
-      out << "\"" << optional.value << "\"";
-    } else {
-      out << optional.value;
-    }
-    out << ")";
-    return out;
-  } else {
-    return out << "XTypes::Optional<" << getTypeName<T>() << ">()";
+  size_t size = 0;
+  OpenDDS::DCPS::serialized_size(OpenDDS::XTypes::get_typeobject_encoding(), size, to);
+  ACE_Message_Block buffer(size);
+  OpenDDS::DCPS::Serializer ser(&buffer, OpenDDS::XTypes::get_typeobject_encoding());
+  if (!(ser << to)) {
+    be_util::misc_error_and_abort("Failed to serialize type object");
   }
-}
-
-std::ostream&
-operator<<(std::ostream& out, const OpenDDS::XTypes::AppliedVerbatimAnnotation& ann)
-{
-  return out
-    << "XTypes::AppliedVerbatimAnnotation("
-    << ann.placement << ", "
-    << ann.language << ", "
-    << ann.text
-    << ")";
-}
-
-std::ostream&
-operator<<(std::ostream& out, const OpenDDS::XTypes::AppliedBuiltinTypeAnnotations& ann)
-{
-  return out
-    << "XTypes::AppliedBuiltinTypeAnnotations("
-    << ann.verbatim
-    << ")";
-}
 
-std::ostream&
-operator<<(std::ostream& out, const OpenDDS::XTypes::NameHash& name_hash)
-{
-  for (size_t i = 0; i < sizeof name_hash; ++i) {
-    out << int(name_hash[i]);
-    if (i < sizeof name_hash - 1) {
-      out << ", ";
+  for (char* ptr = buffer.rd_ptr(); ptr != buffer.wr_ptr(); ++ptr) {
```

Comment:
```suggestion
  for (const char* ptr = buffer.rd_ptr(); ptr != buffer.wr_ptr(); ++ptr) {
```

Prefer adding const here since we're not going to change the data... not a string requirement if it makes the rest of the code clunky or hard to read